extensions/amp-bind/0.1/test/test-bind-integration.js

```diff
@@ -162,6 +162,73 @@ describe.configure().retryOnSaucelabs().run('integration amp-bind', function() {
     });
   });
 
+  describe('amp-live-list integration', () => {
+    function createFromServer(childAttrs = []) {
+      const parent = document.createElement('div');
+      const itemsCont = document.createElement('div');
+      itemsCont.setAttribute('items', '');
+      parent.appendChild(itemsCont);
+      for (let i = 0; i < childAttrs.length; i++) {
+        const childAttr = childAttrs[i];
+        const child = document.createElement('div');
+        child.setAttribute('id', `${childAttr.id}`);
+        child.setAttribute('data-sort-time',
+            `${childAttr.sortTime || Date.now()}`);
+        if ('updateTime' in childAttr) {
+          child.setAttribute('data-update-time', `${childAttr.updateTime}`);
+        }
+        if ('tombstone' in childAttr) {
+          child.setAttribute('data-tombstone', '');
+        }
+        child.innerHTML = '<p [text]="liveListText">unbound</p>';
+        itemsCont.appendChild(child);
+      }
+      return parent;
+    }
+
+    it('should detect bindings in initial live-list elements', () => {
+      const liveListItems = fixture.doc.getElementById('liveListItems');
+      expect(liveListItems.children.length).to.equal(1);
+
+      const liveListItem1 = fixture.doc.getElementById('liveListItem1');
+      expect(liveListItem1.firstElementChild.textContent).to.equal('unbound');
+
+      const button = fixture.doc.getElementById('changeLiveListTextButton');
+      button.click();
+      return waitForBindApplication().then(() => {
+        expect(liveListItem1.firstElementChild.textContent).to
+            .equal('hello world');
+      });
+    });
+
+    it('should apply scope to bindings in new list elements', () => {
+      const liveList = fixture.doc.getElementById('liveList');
+      const liveListItems = fixture.doc.getElementById('liveListItems');
+      expect(liveListItems.children.length).to.equal(1);
+
+      const liveListItem1 = fixture.doc.getElementById('liveListItem1');
+      expect(liveListItem1.firstElementChild.textContent).to.equal('unbound');
+
+      const impl = liveList.implementation_;
+      const update = createFromServer([{id: 'liveListItem2'}]);
+      impl.update(update);
+      fixture.doc.getElementById('liveListUpdateButton').click();
+
+      let liveListItem2;
```

Comment:
I'm usually not a fan this kind of var naming since counters add little semantic value. "Before" and "after" are better modifiers for the context, though there may be even better ones.